site/content/how-to/monitoring/troubleshooting.md

```diff
@@ -65,54 +79,312 @@ LAST SEEN   TYPE      REASON              OBJECT
+3. Each `location` block has headers and directives that configure the NGINX proxy to forward requests to the `/coffee` path correctly, preserving important client information and ensuring compatibility with the upstream server.
+4. The `upstream` block in the given NGINX configuration defines a group of backend servers and configures how NGINX should load balance requests among them.
+
+Now let's check the behaviour when curl request is sent to the `coffee` application:
+
+Matches location /coffee/ block
+
+```shell
+curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffee/hello
+Handling connection for 8080
+Server address: 10.244.0.13:8080
+Server name: coffee-56b44d4c55-hwpkp
+Date: 13/Jun/2024:22:51:52 +0000
+URI: /coffee/hello
+Request ID: 21fc2baad77337065e7cf2cd57e04383
+```
+
+Matches location = /coffee block
+
+```shell
+curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/coffee
+Handling connection for 8080
+Server address: 10.244.0.13:8080
+Server name: coffee-56b44d4c55-hwpkp
+Date: 13/Jun/2024:22:51:40 +0000
+URI: /coffee
+Request ID: 4d8d719e95063303e290ad74ecd7339f
+```
+
+Matches location / block
+
+```shell
+curl --resolve cafe.example.com:$GW_PORT:$GW_IP http://cafe.example.com:$GW_PORT/
+Handling connection for 8080
+<html>
+<head><title>404 Not Found</title></head>
+<body>
+<center><h1>404 Not Found</h1></center>
+<hr><center>nginx/1.25.4</center>
+</body>
+```
+
+**Warning** The configuration may change in future releases.
+
+#### Metrics for Troubleshooting
+
+Metrics can be useful to identify performance bottlenecks and pinpoint areas of high resource consumption within NGINX Gateway Fabric. To setup metrics collection, refer to this [guide]({{< relref "prometheus.md" >}}). The metrics dashboard will help you understand problems with the way NGINX Gateway Fabric is setup or potential issues that could show up with time.
+
+For example, metrics `nginx_reloads_total` and `nginx_reload_errors_total` offer valuable insights into the system's stability and reliability. A high `nginx_reloads_total` value indicates frequent updates or configuration changes, while a high `nginx_reload_errors_total` value suggests issues with the configuration or other problems preventing successful reloads. Monitoring these metrics helps identify and resolve configuration errors, ensuring consistent service reliability.
+
+In such situations, it's advisable to review the logs of both NGINX and NGINX Gateway containers for any potential error messages. Additionally, verify the configured resources to ensure they are in a valid state.
+
+#### Access the N+ Dashboard
+
+If you have NGINX Gateway Fabric installed with NGINX Plus, you can access the N+ dashboard at `http://localhost:8080/dashboard.html`.
+Note that, the port number i.e `8080` matches the port number you have port-forwarded your NGINX Gateway Fabric Pod. For further [details]({{< relref "dashboard.md" >}})
+
+### Common Errors
```

Comment:
We have a `Common Issues` section and `Common Errors` section in this doc. We should either change the name or collapse them.